flexget/plugins/filter/movie_queue.py

```diff
@@ -396,3 +415,156 @@ def queue_get(session=None, downloaded=False):
+movie_queue_parser.add_argument('page', type=int, required=False, default=1, help='Page number')
+movie_queue_parser.add_argument('max', type=int, required=False, default=50, help='Movies per page')
+movie_queue_parser.add_argument('downloaded', type=bool, required=False, default=False, help='Show only downloaded')
+
+
+@movie_queue_api.route('/')
+class MovieQueueListAPI(APIResource):
+    @api.response(404, 'Page does not exist')
+    @api.response(200, 'Movie queue retrieved successfully', movie_queue_schema)
+    @api.doc(parser=movie_queue_parser)
+    def get(self, session=None):
+        """ List queued movies """
+        args = movie_queue_parser.parse_args()
+        page = args['page']
+        max_results = args['max']
+        downloaded = args['downloaded']
+
+        movie_queue = queue_get(session=session, downloaded=downloaded)
+        count = len(movie_queue)
+
+        pages = int(ceil(count / float(max_results)))
+
+        movie_items = []
+
+        if page > pages and pages != 0:
+            return {'error': 'page %s does not exist' % page}, 404
+
+        start = (page - 1) * max_results
+        finish = start + max_results
+        if finish > count:
+            finish = count
+
+        for movie_number in range(start, finish):
+            movie_items.append(movie_queue[movie_number])
+
+        return jsonify({
+            'movies': [movie.to_dict() for movie in movie_items],
+            'number_of_movies': count,
+            'page_number': page,
+            'total_number_of_pages': pages
+        })
+
+
+movie_add_results_schema = {
+    'type': 'object',
+    'properties': {
+        'message': 'Successfully added movie to movie queue',
+        'movies': {
+            'type': 'object',
+            'properties': {
+                'added': {'type': 'string'},
+                'downloaded': {'type': 'string'},
+                'entry_original_url': {'type': 'string'},
+                'entry_title': {'type': 'string'},
+                'entry_url': {'type': 'string'},
+                'id': {'type': 'integer'},
+                'imdb_id': {'type': 'string'},
+                'quality': {'type': 'string'},
+                'quality_req': {'type': 'string'},
+                'title': {'type': 'string'},
+                'tmdb_id': {'type': 'string'},
+            }
+        }}
+}
+
+movie_add_input_schema = {
+    'type': 'object',
+    'properties': {
+        'title': {'type': 'string'},
+        'imdb_id': {'type': 'string', 'pattern': r'tt\d{7}'},
+        'tmdb_id': {'type': 'integer'},
+        'quality': {'type': 'string', 'format': 'quality_requirements'}
+    },
+    'anyOf': [
+        {'required': ['title']},
+        {'required': ['imdb_id']},
+        {'required': ['tmdb_id']}
+    ]
+}
+
+movie_add_results_schema = api.schema('movie_add_results', movie_add_results_schema)
+movie_add_input_schema = api.schema('movie_add_input_schema', movie_add_input_schema)
+
+
+@movie_queue_api.route('/add/')
```

Comment:
perhaps this should be added to the above as a post.. so rather then /add it should listen on /